### ISSUE TITLE
FYI

### ISSUE THREAD

**Issue by chicagogrooves** [arsnebula/reactive-promise#1]
Interestingly, we started our two projects only a day apart! 
https://github.com/chicagogrooves/deanius-meteor-promise

Feel free to discuss..

**Comment by chrismbeckett** [comment]
In the last Meteor DevShip, there was a lightning talk on [Reactive-Method](https://atmospherejs.com/simple/reactive-method) that also seems to have been published very recently.

Reactive-Method is useful because, at least in Meteor, calling a method is probably what you will be doing async from the client most of the time, and needing to wait on it to complete is a great function for reactivity. The drawback of the package is that it only provides a reactive wrapper around a single async call, and it only does it for one function, a Meteor method call. He is now suggesting he will build other *single purpose* packages for other aynch tasks (like HTTP). This approach could end up being a lot of packages...

Promises, whether using the jQuery.Deferred or your package is much better than callbacks, and has the advantage of waiting on multiple async tasks to complete, and provides a cleaner syntax than nested callbacks.

My package was really designed to provide a bridge around both, providing the ability to wrap one, or multiple promises to a reactive variable. It requires a few more lines of code than Reactive-Method, allows you to wait on multiple Meteor calls, and I specifically made it work with Iron-Router as well.

**Comment by chicagogrooves** [comment]
Interesting, that clears things up a bit. 

My package initially just used jQuery promises, but now abstracts to allow for Q if that promise flavor exists.

In your project, I like the idea of Router integration, and your implementation is straightforward - a map of task names to `ReactiveVar` that get `set` upon promise resolution.

I wonder if naming the tasks is absolutely necessary, or if some auto-generated value could be used, so that you could check the ready status of the promise itself - does that make sense ? I think you're going in the right direction, and I'm curious to see what the `ReactiveMethod` author will come up with next.
